### PR TITLE
[AIRFLOW-3944] Remove code smells

### PR DESCRIPTION
**Pull request by BasPH**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3944
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

I removed various code smells:

- Removed mutable default arguments
- Replace function calls by literals
- Remove reusing same variable names
- Make function name lowercase
- Remove redundant parentheses
- Remove unreachable code

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/dataproc_operator.py [hunk 1]
```diff
@@ -769,7 +769,7 @@ def __init__(
         self.dataproc_properties = dataproc_pig_properties
         self.dataproc_jars = dataproc_pig_jars
         self.region = region
-        self.job_error_states = job_error_states
+        self.job_error_states = job_error_states if job_error_states is not None else ['ERROR']
```
Comment (Fokko) [inline airflow/contrib/operators/dataproc_operator.py]: Nice one. While we're at it, how about making this a set as well? I expect to error states to be unique.

**Review by Fokko — COMMENTED** [review]
Thanks @BasPH for picking this up, one comment that I would like to have your opinion on 👍

File: airflow/contrib/operators/dataproc_operator.py (hunk 1 above)
Comment (BasPH) [reply airflow/contrib/operators/dataproc_operator.py]: Good catch. I don't see any reason for duplicates in it, so set is better.

I scanned through `airflow.contrib.hooks.gcp_dataproc_hook.py` and there's nothing requiring/checking list type, so I don't see any problems and will replace all `job_error_states` by sets.

Comment (BasPH) [reply airflow/contrib/operators/dataproc_operator.py]: @Fokko I pushed the changes, CI is rerunning.

**Review by Fokko — APPROVED** [review]
Thanks, looking good. Waiting for the CI to pass.

**Pull request merged**